### PR TITLE
chore: review error handling and remove unwraps 

### PR DESCRIPTION
**Pull request by marcospb19-cw**
### **User description**
In this PR I try removing unwraps, or, when not possible, replacing them by expect with a message of why that isn't going to fail.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Improved error handling across multiple modules by replacing `unwrap()` calls with proper error handling mechanisms.
- Enhanced the `ExternalTransaction` struct to return `Result<BlockNumber>` instead of unwrapping.
- Implemented `TryFrom` trait for `SubscriptionMessage` conversion to handle potential serialization errors.
- Simplified and optimized various utility functions and methods, including JSON handling and type manipulations.
- Improved RPC logging by using pretty-printed JSON strings.
- Enhanced test suite by using temporary directories and removing unnecessary file system operations.
- Refactored JSON object manipulations in Redis storage for better efficiency.
- Simplified `InMemoryHistory` and `ExecutionConflictsBuilder` implementations.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Error handling</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>miner.rs</strong><dd><code>Enhance error handling in miner module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/miner/miner.rs

<li>Replaced unwraps with error handling in <code>block_number()</code> call<br> <li> Improved error handling in time-related operations<br> <li> Removed unnecessary print statement<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-16f448afc7dae3e8e802f676a86dbd7051e19f6c17cbbeb53eb730d726372629">+7/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>execution.rs</strong><dd><code>Enhance error handling in EvmExecution</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/primitives/execution.rs

<li>Improved error handling in <code>EvmExecution</code> implementation<br> <li> Added explanatory comments for error handling<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-cd13932ed3cbc97c18547275105d872a204eda4ad4bf0c2811b18553b48e9038">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>external_transaction.rs</strong><dd><code>Improve error handling in ExternalTransaction</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/primitives/external_transaction.rs

<li>Changed <code>block_number()</code> method to return <code>Result<BlockNumber></code><br> <li> Added context to error handling<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-a5bcf577600cee6a4844b4015797a92684df8ec490f06f66b4a556881a66032d">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>log_mined.rs</strong><dd><code>Implement TryFrom for SubscriptionMessage conversion</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/primitives/log_mined.rs

<li>Changed <code>From</code> implementation to <code>TryFrom</code> for <code>SubscriptionMessage</code><br> <li> Added error handling for JSON serialization<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-0ce210a69394a14d9a6424cf5ddf8b84c10a535b9ecae1dd34c0dbcff4beaf07">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rpc_subscriptions.rs</strong><dd><code>Improve error handling in RPC subscriptions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/rpc/rpc_subscriptions.rs

<li>Enhanced <code>notify</code> function to handle conversion errors<br> <li> Implemented generic error handling for message conversion<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-a07ba1157cec6586a0520b8dfb11e9d073359af24471ac286ff5eaa2669d355c">+14/-2</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>code_hash.rs</strong><dd><code>Remove unused CodeHash conversion</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/primitives/code_hash.rs

- Removed unnecessary `From<Vec<u8>>` implementation for `CodeHash`

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-588f5e271548cdb222c974e0c5b540e707b96f69ce5f3d7bd36b9e0572f692d8">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>execution_conflict.rs</strong><dd><code>Simplify ExecutionConflictsBuilder</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/primitives/execution_conflict.rs

- Simplified `build()` method in `ExecutionConflictsBuilder`

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-989664061f5331343a7fe685e7bbc19f01044105c3e966afd41233c163671df1">+1/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rpc_middleware.rs</strong><dd><code>Improve RPC logging format</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/rpc/rpc_middleware.rs

- Changed `to_json_value` to `to_json_string_pretty` for logging

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-31b7182aec4416845e60ec9ec16720ae97d31260e1a9c50d601d542dee8776f5">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>inmemory_history.rs</strong><dd><code>Simplify InMemoryHistory reset logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/storage/inmemory/inmemory_history.rs

- Simplified `reset_at` method using `NonEmpty::from_vec`

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-322061156099977548ec07fa5d628c583d880f23259fdb3a53fd4e2ad9a21789">+2/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>redis_permanent.rs</strong><dd><code>Refactor JSON handling in Redis storage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/storage/redis/redis_permanent.rs

<li>Replaced <code>to_json_value</code> with <code>to_json_object</code> for account modification<br> <li> Simplified JSON object manipulation<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-068ece78c4a374a891db1f4759a7a035df7aa981bf04c3bffecfd44e94f5841b">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rocks_state.rs</strong><dd><code>Improve RocksDB state handling and testing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/storage/rocks/rocks_state.rs

<li>Simplified <code>db_path_filename</code> method<br> <li> Updated test to use temporary directory<br> <li> Removed unnecessary file system cleanup<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-f7b822022d2c4fd93ec507cd6b5302dcf1646acbf0ee0dd077a047272b686009">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ext.rs</strong><dd><code>Enhance JSON handling and type utilities</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/ext.rs

<li>Enhanced <code>type_basename</code> function to handle edge cases<br> <li> Added <code>to_json_object</code> function for JSON object serialization<br>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/1624/files#diff-4ff673578dd06edb71da19750f05cb48bbbd3c78ded3dbeda59409354c35cef6">+12/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/eth/rpc/rpc_middleware.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ impl<'a> RpcServiceT<'a> for RpcMiddleware {
             rpc_client = %client,
             rpc_id = %request.id,
             rpc_method = %method,
-            rpc_params = %to_json_value(&request.params),
+            rpc_params = %to_json_string_pretty(&request.params),
```
Comment (dinhani-cw) [inline src/eth/rpc/rpc_middleware.rs:106]: Pretty string is formatted with multiple lines and will break logging.

**Comment by dinhani-cw** [comment]
Please break this PR in multiple PRs.
There are too many changes happening and it is hard to track which ones are not introducing bugs because there were a lot of rewrites.

File: src/eth/rpc/rpc_middleware.rs (hunk 1 above)
Comment (marcospb19-cw) [reply src/eth/rpc/rpc_middleware.rs]: Good catch, I was supposed to call to_json_string and not to_json_string_pretty.

**Comment by marcospb19-cw** [comment]
> There are too many changes happening and it is hard to track which ones are not introducing bugs because there are a lot of rewrites.

I separated each modification into a separate commit so that it's possible to review them individually.

**Pull request merged**